R/make_split_fun.R

```diff
@@ -94,27 +96,35 @@ validate_split_result <- function(pinfo, component = NULL) {
 #' list is as the rtables internals expect it, most of which are not
 #' relevant to end users.
 #'
+#' @note Column splitting will not work correctly if a split function
+#' calls `make_split_result` without specifying subset expressions;
+#' row splitting will work as normal. This is due to the fact that
+#' subsetting expressions are used during column splitting to
+#' represent the data associated with facets, while actual data
+#' subsets are used during row splitting.
 #'
 #' @examples
 #' splres <- make_split_result(
 #'   values = c("hi", "lo"),
 #'   datasplit = list(hi = mtcars, lo = mtcars[1:10, ]),
-#'   labels = c("more data", "less data")
+#'   labels = c("more data", "less data"),
+#'   subset_exprs = list(expression(TRUE), expression(seq_along(wt) <= 10))
```

Comment:
so this would do a subset only in the column space, while it is ignored in the row space, right?

Reply:
Correct. Faceting in column space is tracked via expressions, while faceting in row space is immediately materialized during tabulation. This is largely for a combination of performance and legacy reasons.